src/xtgeo/grid3d/_ecl_grid.py

```diff
@@ -357,22 +376,38 @@ class EclGrid(ABC):
     means active, 2 means rock volume only, 3 means pore volume only.
     """
 
-    coord: np.ndarray
-    zcorn: np.ndarray
-    actnum: Optional[np.ndarray] = None
+    @property
+    @abstractmethod
+    def coordinates(self) -> np.ndarray:
```

Comment:
Is it fruitful to rename `coord` to `coordinates`? I agree that it is beneficial to be explicit, but we'll introduce more api breaking than necessary, and we'll have both `coordinates` and `coord` as parameter names (e.g.`xtgeo_coord`). Similar with `activity_number` - where `actnum` stands for `active_number`.

We could instead include in the docstring that `zcorn` is corner height etc.

Reply:
I am open to suggestions here, but I needed a different name for the abstract property which gives you the coord array as having these properties coinside for dataclasses is non-trivial. Someone created a package to solve it but I was skeptical of adding the dependency: https://github.com/MichaelSchneeberger/dataclass-abc .

Reply:
We discussed a bit and concluded that there were two options: Use a normal class instead of a dataclass to avoid the name clash, or prefix the variables in the dataclass with an underscore, however that will be reflected in the signature. I think we were leaning towards using a regular class though

Reply:
08594bf81af53d668491798c4f04d6d6b608c18c should commit the changes we discussed.